fix gst log formatting arguments

ystreet opened the pull request:
passing a log string in a printf format specifier can do bad things if
e.g the message contains printf arguments as may occur when logging
random data received from e.g. libcurl.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

ystreet committed:
fix gst log formatting arguments

passing a log string in a printf format specifier can do bad things if
e.g the message contains printf arguments as may occur when logging
random data received from e.g. libcurl.

louisharris commented:
Hey, I noticed this build failed due to an older python version being used in the build, I have a change ready to up the version which passes the build. After this change is merged, this build should pass.

louisharris commented:
Could you resubmit the PR after pulling the new changes merged? The TravisCI build should pass then.

ystreet committed:
build: also declare a dep on the amazon aws sdk

Otherwise we may get linking errors if the amazon sdk is installed
outside the regular prefix like:

FAILED: tests/check/s3sink
c++  -o tests/check/s3sink tests/check/s3sink.p/s3sink.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group src/libgsts3elements.so /usr/lib64/libgstreamer-1.0.so /usr/lib64/libgobject-2.0.so /usr/lib64/libglib-2.0.so /usr/lib64/libgstbase-1.0.so /usr/lib64/libgstcheck-1.0.so -lm -lrt -Wl,--end-group '-Wl,-rpath,$ORIGIN/../../src' -Wl,-rpath-link,/path/to/amazon-s3-gst-plugin/build/src
/usr/bin/ld: warning: libaws-cpp-sdk-s3.so, needed by src/libgsts3elements.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libaws-cpp-sdk-core.so, needed by src/libgsts3elements.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libaws-cpp-sdk-sts.so, needed by /path/to/amazon-s3-gst-plugin/build/src/libgstawscredentials-1.0.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: src/libgsts3elements.so: undefined reference to `Aws::InitAPI(Aws::SDKOptions const&)'
/usr/bin/ld: src/libgsts3elements.so: undefined reference to `Aws::S3::Model::CompletedPart::CompletedPart()'
/usr/bin/ld: src/libgsts3elements.so: undefined reference to `Aws::S3::S3Client::~S3Client()'
...
